### PR TITLE
Re-enable some linting

### PR DESCRIPTION
**Pull request by erikjohnston**
These were disabled in #14633

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ line-length = 88
 #  E731: do not assign a lambda expression, use a def
 #  E501: Line too long (black enforces this for us)
 #
-# See https://github.com/charliermarsh/ruff/#pyflakes
+# See https://github.com/charliermarsh/ruff/#pyflakes-f
```
Comment (clokep) [inline pyproject.toml:51]: If we're removing these from the ignore list, let's nuke the comment too!

File: synapse/config/_base.pyi [hunk 2]
```diff
@@ -20,7 +18,7 @@ from typing import (
 
 import jinja2
 
-from synapse.config import (
+from synapse.config import (  # noqa: F401
```
Comment (clokep) [inline synapse/config/_base.pyi:21]: Do you need to ignore on both the import and usage line?

Comment (erikjohnston) [reply synapse/config/_base.pyi:21]: No, we just need it on the import line? Each `noqa` comment is suppressing different errors?

Comment (clokep) [reply synapse/config/_base.pyi:21]: TIL that 401 != 881. (I swear those all said 401...)

**Review by clokep — APPROVED** [review]

**Pull request merged**